api/src/main/java/org/openmrs/CohortMembership.java

```diff
@@ -99,6 +118,20 @@ public void setEndDate(Date endDate) {
 
 	@Override
 	public int compareTo(CohortMembership o) {
-		return this.getPatient().getPatientId() - o.getPatient().getPatientId();
+		if (Objects.equals(this.getPatient().getPatientId(), o.getPatient().getPatientId())
+				&& Objects.equals(this.getCohort().getCohortId(), o.getCohort().getCohortId())
+				&& this.getStartDate().equals(o.getStartDate())
```

Comment:
Prefer to use OpenmrsUtil.compare method for dates, since sometimes what you get back from hibernate is a java.sql.Date which is not directly comparable to a java.util.Date.